File: arrays_and_hashing/contains_duplicate.py
# Memory: 8.2 MB•Time: 45ms•Submitted at: 04/08/2026 13:27
# Beats 7.01% in runtime, 2.26% in memory

# Given an integer array nums, return true if any value appears more than once in the array, otherwise return false.

class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        # Map approach, if value exists return false

        numbers = []

        for i in range(len(nums)):
            if nums[i] in numbers:
                return True
            else:
                numbers.append(nums[i])
        
        return False

File: arrays_and_hashing/contains_duplicate.cpp
// Memory: 150.8 MB•Time: 2ms•Submitted at: 04/08/2026 13:46
// Beats 93.17% in runtime, 5.77% in memory

#include <map>
#include <algorithm>

using namespace std;

class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        // Use map to track occurences of values in array
        // Return true if any value of map increments to 2

        map<int, int> numbers;
        bool result = false;

        for (int num : nums) {
            if (numbers.contains(num)) {
                result = true;
                break;
            } else {
                numbers[num] = 1;
            }
        }

        return result;
    }
};
